### DISCUSSION TITLE
OpenTelemetry Integration

### DISCUSSION THREAD

**Discussion by 6543** [woodpecker-ci/woodpecker#3433] [General]
* [ ] trace of a build from webhook to send status to forge back
* [ ] track agent initialisation ...
* [ ] track time to pull images ...
* [ ] inject a uuid into steps to track int future within steps ..
* [ ] ...

**Comment by 6543** [comment]
https://opentelemetry.io/  
https://github.com/open-telemetry/opentelemetry-go  
https://opentelemetry.io/docs/instrumentation/go/getting-started/

**Comment by schewara** [comment]
This could then also replace the Prometheus integration once the Metriks SDK is GA
